app/src/js/controllers/ImageController.js

```diff
@@ -47,6 +48,44 @@ angular.module('media_manager')
       CourseCache.current_image = CourseCache.images[ic.index];
       resetBreadcrumb();
     }
-  }
+  };
+
+  ic.newLabel = '';
+  ic.newValue = '';
+  ic.saveMetadata = function(label, value){
+    if(label !== undefined && value !== undefined){
+      ic.image.metadata.forEach(function(item, index, arr){
+        if(item.label == label){
+          arr[index].label = label;
+          arr[index].value = value;
+          Image.update({}, ic.image, function success(data){
```

Comment:
I don't quite understand the purpose of this `forEach` on the metadata array. Why does it need to iterate over the metadata array, when you could just pass the index into the function and lookup the label/value pair directly? There could be multiple pairs in the metadata array with the same label, and this code would match and overwrite all such pairs it seems.